src/EFCore.PG/Query/Sql/Internal/NpgsqlQuerySqlGenerator.cs

```diff
@@ -198,6 +209,17 @@ protected override Expression VisitBinary(BinaryExpression expression)
             }
         }
 
+        /// <inheritdoc />
+        protected override Expression VisitDefault(DefaultExpression e)
```

Comment:
What is this required for exactly?

Reply:
Passing back a `DefaultExpression` throws if we don't override. The expression itself is introduced because `lower` and `upper` return `null` on empty or infinite bounds. But since our range types may have non-nullable bounds (e.g. `NpgsqlRange<int>.LowerBound` is `int` not `int?`) we need to coalesce the database `null` to what the CLR bound ought to be (currently, the default value).

Needing to do this might be another symptom of `NpgsqlRange<T>` needing some more work... but changes there would be breaking, so my thought was that mitigating the incompatibility here would be reasonable for now.

Reply:
The rest of this PR is totally fine, this is the only part of this PR that I want to understand a bit better (and have no time today), will look again tomorrow.

Reply:
@roji Did you still want to take a look at this?